smoke_zephyr/configuration.py

```diff
@@ -202,6 +202,63 @@ def save(self):
 		with open(self.configuration_file, 'w') as file_h:
 			file_h.write(self._serializer('dumps', self._storage))
 
+class MemoryConfiguration(Configuration):
+	"""
+	This class provides a generic object for parsing information from variables in multiple formats.
+	"""
+
+	def __init__(self, mem_object, object_type, prefix=''):
+		"""
+		:param str mem_object: The memory object to parse.
+		:param str prefix: String to be prefixed to all option names.
+		:param str object_type: String to identify type of serializer to use.
+		"""
+		self.prefix= prefix
+		self.seperator = '.'
+		self.object_type = object_type
+		if type(mem_object) != 'dict':
```

Comment:
This should be isinstance and really shouldn't even be checking if it's a dict but rather if it will correctly respond to `__getitem__`.